src/get-sibling-tiles-and-offsets.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const { DEFAULT_LENGTH, TILE_SIZE } = require('../utils/constants');
+const getProximateCenterIndex = require('./get-proximate-center-index');
+const getIndexRelativeToCenter = require('./get-index-relative-to-center');
+const getRelativePx = require('./get-relative-pixel-offsets');
+
+function getSiblingTilesAndOffsets(
+  pxCoordinates,
+  zoom,
+  length = DEFAULT_LENGTH,
+  tileSize = TILE_SIZE
+) {
+  const dimensions = { h: length, w: length };
+  const centerIndex = getProximateCenterIndex(pxCoordinates, zoom, tileSize);
+  const topLeftIndex = getIndexRelativeToCenter({ x: 0, y: 0 }, centerIndex, dimensions, tileSize);
+  const bottomRightIndex = getIndexRelativeToCenter({ x: length, y: length }, centerIndex, dimensions, tileSize);
+  const maxTiles = Math.pow(2, zoom);
+
+  const imgProperties = {};
+  imgProperties.tiles = [];
+
+  // iterate through our grid & determine what additional tiles are necessary
+  // in order to make the final image
+  for (let row = topLeftIndex.row; row <= bottomRightIndex.row; row++) {
+    if (row < 0) continue;
+    if (row >= maxTiles) break;
```

Comment:
Would this just be `>` to factor the `maxTiles`?

Reply:
Yep, good catch.

Reply:
Hmmm, even though tests are passing in my latest commit, I'm seeing some weird false-positive behavior for the get-sibling-tiles-and-offsets tests now: 

![Screen Shot 2020-04-29 at 12 03 17 PM](https://user-images.githubusercontent.com/16272584/80618852-ed633380-8a11-11ea-9655-435ef1b7aea8.png)
[link](https://travis-ci.com/github/mapbox/quilt/jobs/325085201)

Reply:
I'll need to dig into this a bit more to understand why this change is causing that behavior, but I suspect it's primarily to do with how the test is written since the change lines did not modify how `zoom` values are handled at all.

Reply:
Ah, yep - bad fixture. Fixed in 4d7622b